app/routes/login/route.tsx

```diff
@@ -0,0 +1,23 @@
+import { LoaderFunctionArgs, redirect } from '@remix-run/node';
+import { commitSession, getSession, initializeEcomApiForRequest } from '~/lib/ecom/session';
+
+export async function loader({ request }: LoaderFunctionArgs) {
+    const currentUrl = new URL(request.url);
+    const returnUrlPath = currentUrl.searchParams.get('returnPath') ?? '/';
```

Comment:
Why don't we always return them to the page from which they've logged in?

Reply:
they not always log in from home page. they might click login from any other public page, e.g. category, product details. it is better to redirect back to same page.
returnPath is set in query params when user click login button

Reply:
This is exactly what I'm saying. Why don't we always redirect them to where they came from instead of manually setting return path? Do we not have the referrer header?

Reply:
this is exactly what we are doing here. but idea with referrer is better, thanks, will change this

Reply:
switch to `referer` header